algorithms/linfa-clustering/src/dbscan/algorithm.rs

```diff
@@ -142,62 +202,27 @@ impl<F: Float, D: Data<Elem = F>, T>
     }
 }
 
-impl<F: Float, D: Data<Elem = F>> Transformer<&ArrayBase<D, Ix2>, Array1<Option<usize>>>
-    for DbscanHyperParamsBuilder<F>
-{
-    fn transform(&self, observations: &ArrayBase<D, Ix2>) -> Array1<Option<usize>> {
+impl<F: Float, DF: 'static + Distance<F>, N: NearestNeighbour> DbscanHyperParamsBuilder<F, DF, N> {
```

Comment:
distinguishing between params and a param builder does not make so much sense here 'cause build does not fail atm, right?

Reply:
`build` does panic if the builder contains invalid parameters, but we can move those checks elsewhere during initialization. Might make sense to get rid of the builder altogether and have the hyperparams act as the builder.

Reply:
it's easier without the build function call, perhaps just move the build call into prediction